src/main/java/ca/bc/gov/iamp/bcparis/message/MessageUtils.java

```diff
@@ -34,23 +35,23 @@
     }};
 
     public static String GetValue(String message, String key) {
+
         if (StringUtils.isEmpty(message)) return null;
-        message = removeKnownEnd(message);
 
-        HashSet<String> knownTokens = buildKnownTokens(key);
+        message = removeKnownEnd(message);
 
         int startIndex = message.indexOf(key + SEMICOLLON);
         if (startIndex == -1) return null;
 
         startIndex += key.length() + 1;
 
-        int currentEndIndex = message.length();
+        message = message.substring(startIndex);
 
-        for (String token : knownTokens) {
+        int currentEndIndex = message.length();
 
-            int tokenIndex = message.indexOf(token + SEMICOLLON);
+        for (String token : KNOWN_TOKENS) {
```

Comment:
I'm assuming this is like a javascript forEach, my java syntax knowledge is rusty. Will this ever reach an out of bounds?

Reply:
yes it's a foreach loop